lasy/utils/laser_utils.py

```diff
@@ -205,3 +207,165 @@ def get_full_field(laser, theta=0, slice=0, slice_axis="x", Nt=None):
+        - 'rt' : The laser pulse is represented on a 2D grid:
+                 Cylindrical (r) transversely, and temporal (t) longitudinally.
+
+    is_envelope : bool (optional)
+        Whether the field provided uses the envelope representation, as used
+        internally in lasy. If False, field is assumed to represent the
+        electric field.
+
+    omega0 : scalar
+        Angular frequency at which the envelope is defined.
+        Required if an only if is_envelope is True.
+
+    is_hilbert : boolean (optional)
+        If True, the field argument is assumed to be a Hilbert transform, and
+        is used through the computation. Otherwise, the Hilbert transform is
+        calculated in the function.
+
+    phase_unwrap_1d : boolean (optional)
+        Whether the phase unwrapping is done in 1D.
+        This is not recommended, as the unwrapping will not be accurate,
+        but it might be the only practical solution when dim is 'xyt'.
+
+    lower_bound : scalar (optional)
+        Relative lower bound for the local frequency
+        Frequencies lower than lower_bound * central_omega are cut
+        to lower_bound * central_omega.
+
+    upper_bound : scalar (optional)
+        Relative upper bound for the local frequency
+        Frequencies larger than upper_bound * central_omega are cut
+        to upper_bound * central_omega.
+
+    Returns
+    -------
+    omega : nd array of doubles
+        local angular frequency.
+
+    central_omega : scalar
+        Central angular frequency (averaged omega, weighted by the local
+        envelope amplitude).
+    """
+    # Assumes t is last dimension!
+    if is_envelope:
+        assert omega0 is not None
+        phase = np.unwrap(np.angle(grid.field))
+        omega = omega0 + np.gradient(-phase, grid.axes[-1], axis=-1, edge_order=2)
+        central_omega = np.average(omega, weights=np.abs(grid.field))
+    else:
+        assert dim in ["xyt", "rt"]
+        if dim == "xyt" and not phase_unwrap_1d:
+            print("WARNING: using 3D phase unwrapping, this can be expensive")
+
+        if not is_hilbert:
+            h = np.squeeze(hilbert(grid.field))
+        else:
+            h = np.squeeze(grid.field)
+        if phase_unwrap_1d:
+            phase = np.unwrap(np.angle(h))
+        else:
+            phase = unwrap_phase(np.angle(h))
+        omega = np.gradient(-phase, grid.axes[-1], axis=-1, edge_order=2)
+
+        if dim == "xyt":
+            weights = np.abs(h)
+        else:
+            r = grid.axes[0].reshape((grid.axes[0].size, 1))
+            weights = r * np.abs(h)
+        central_omega = np.average(omega, weights=weights)
+
+    # Filter out too small frequencies
+    omega = np.maximum(omega, lower_bound * central_omega)
+    # Filter out too large frequencies
+    omega = np.minimum(omega, upper_bound * central_omega)
+
+    return omega, central_omega
+
+
+def field_to_a0(grid, omega0):
```

Comment:
Same comment as above. Maybe `a0` can be misleading. `field_to_potential` could also work, for example. Not a big issue in any case.